SimpleITK/utilities/vtk.py

```diff
@@ -64,63 +63,52 @@ def sitk2vtk(img, debugOn=False):
             1.0,
         ]
 
+    # Create VTK image and set its metadata
+    vtk_image = vtk.vtkImageData()
     vtk_image.SetDimensions(size)
     vtk_image.SetSpacing(spacing)
     vtk_image.SetOrigin(origin)
     vtk_image.SetExtent(0, size[0] - 1, 0, size[1] - 1, 0, size[2] - 1)
-
     if vtk.vtkVersion.GetVTKMajorVersion() < 9:
-        print("Warning: VTK version <9.  No direction matrix.")
+        logging.warning(
+            "VTK version <9 does not support direction matrix which is ignored"
+        )
     else:
         vtk_image.SetDirectionMatrix(direction)
 
-    depth_array = numpy_support.numpy_to_vtk(i2.ravel())
+    # Set pixel data
+    depth_array = vtknp.numpy_to_vtk(sitk.GetArrayViewFromImage(image).ravel())
     depth_array.SetNumberOfComponents(ncomp)
     vtk_image.GetPointData().SetScalars(depth_array)
 
     vtk_image.Modified()
-    if debugOn:
-        print("Volume object inside sitk2vtk")
-        print(vtk_image)
-        print("num components = ", ncomp)
-        print(size)
-        print(origin)
-        print(spacing)
-        print(vtk_image.GetScalarComponentAsFloat(0, 0, 0, 0))
-
     return vtk_image
 
 
-def vtk2sitk(vtkimg, debug=False):
-    """Takes a VTK image, returns a SimpleITK image."""
-    sd = vtkimg.GetPointData().GetScalars()
-    npdata = vtknp.vtk_to_numpy(sd)
+def vtk2sitk(image: vtk.vtkImageData) -> sitk.Image:
+    """Convert a VTK image to a  SimpleITK image.
 
-    dims = list(vtkimg.GetDimensions())
-    origin = vtkimg.GetOrigin()
-    spacing = vtkimg.GetSpacing()
+    :param image: Image to convert.
+    :type image: vtk.vtkImageData
+    :return: A SimpleITK image.
+    :rtype: sitk.Image
+    """
+    sd = image.GetPointData().GetScalars()
+    npdata = vtknp.vtk_to_numpy(sd)
+    dims = list(image.GetDimensions())
+    dims.reverse()
+    npdata.shape = tuple(dims)
 
-    if debug:
-        print("dims:", dims)
-        print("origin:", origin)
-        print("spacing:", spacing)
+    sitk_image = sitk.GetImageFromArray(npdata)
+    sitk_image.SetSpacing(image.GetSpacing())
+    sitk_image.SetOrigin(image.GetOrigin())
+    # By default, direction is identity.
 
-        print("numpy type:", npdata.dtype)
-        print("numpy shape:", npdata.shape)
+    if vtk.vtkVersion.GetVTKMajorVersion() >= 9:
+        # Copy the direction matrix into a list
+        dir_mat = image.GetDirectionMatrix()
+        direction = [0] * 9
+        dir_mat.DeepCopy(direction, dir_mat)
```

Comment:
Not sure why this change was made here too.

Reply:
Following the approach [taken in Slicer](https://github.com/Slicer/Slicer/blob/e53e8af9c4a0b60adee28b5eca5fc1b5ff2da9ea/Base/Python/slicer/util.py#L1129-L1130). Code is more concise.